src/lib/renderComponent.tsx

```diff
@@ -83,7 +73,11 @@ const renderComponent = <P extends {}>(
 
         // Resolve styles using resolved variables, merge results, allow props.styles to override
         const mergedStyles = mergeComponentStyles(componentStyles[displayName], props.styles)
-        const styleParam: ComponentStyleFunctionParam = { props, variables: resolvedVariables }
+        const appliedProps = { ...state, ...props }
```

Comment:
👍

Reply:
one note - before we'll come up with a name for this 'merged' object, maybe we could use something that is a bit dumb but explicit and descriptive - like `stateAndProps`?

Reply:
done, thanks